### ISSUE TITLE
Freqtrade edge completely wrong results

### ISSUE THREAD

**Issue by Michele-GitHub** [freqtrade/freqtrade#2685]
## Step 1: Have you search for this issue before posting it?

If you have discovered a bug in the bot, please [search our issue tracker](https://github.com/freqtrade/freqtrade/issues?q=is%3Aissue). 
If it hasn't been reported, please create a new issue.

## Step 2: Describe your environment

  * Operating system: Ubuntu 19.10
  * Python Version: 3.7.5 
  * CCXT version: ccxt==1.20.46
  * Branch: Develop
  * Last Commit ID: 703924d6c4a39db57576763e7d51376d707460e9
  
## Step 3: Describe the problem:

freqtrade edge results are totally different from freqtrade backtesting results. The test is explained below:

### Steps to reproduce:

Example strategy :
``` python
class ProvaEdge(IStrategy):
    INTERFACE_VERSION = 2
    minimal_roi = {"0": 100}
    stoploss = -0.114
    trailing_stop = False
    ticker_interval = '1h'
    process_only_new_candles = False
    use_sell_signal = True
    sell_profit_only = False
    ignore_roi_if_buy_signal = False
    startup_candle_count: int = 20
    order_types = {
        'buy': 'limit',
        'sell': 'limit',
        'stoploss': 'market',
        'stoploss_on_exchange': True}
    order_time_in_force = {
        'buy': 'gtc',
        'sell': 'gtc'}

    def informative_pairs(self):
        return []

    def populate_indicators(self, dataframe: DataFrame, metadata: dict) -> DataFrame:

        # Bollinger bands
        bollinger = qtpylib.bollinger_bands(qtpylib.typical_price(dataframe), window=20, stds=2)
        dataframe['bb_lowerband'] = bollinger['lower']
        dataframe['bb_middleband'] = bollinger['mid']

        return dataframe

    def populate_buy_trend(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
        dataframe.loc[
            (
                (dataframe['close'] < dataframe['bb_lowerband'])
            ),
            'buy'] = 1

        return dataframe

    def populate_sell_trend(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
        dataframe.loc[
            (
                (dataframe['close'] > dataframe['bb_middleband'])
            ),
            'sell'] = 1
        return dataframe
```

ConfigProvaEdge.json :

``` json 
{
    "max_open_trades": 100,
    "stake_currency": "BTC",
    "stake_amount": 0.01,
    "fiat_display_currency": "EUR",
    "ticker_interval" : "1h",
    "dry_run": true,
    "trailing_stop": false,
    "unfilledtimeout": {
        "buy": 10,
        "sell": 30
    },
    "bid_strategy": {
        "ask_last_balance": 0.0,
        "use_order_book": false,
        "order_book_top": 1,
        "check_depth_of_market": {
            "enabled": false,
            "bids_to_ask_delta": 1
        }
    },
    "ask_strategy":{
        "use_order_book": false,
        "order_book_min": 1,
        "order_book_max": 9,
        "use_sell_signal": true,
        "sell_profit_only": false,
        "ignore_roi_if_buy_signal": false
    },
    "exchange": {
        "name": "binance",
        "key": "xxxx",
        "secret": "yyy",
        "ccxt_config": {"enableRateLimit": true},
        "ccxt_async_config": {
            "enableRateLimit": true,
            "rateLimit": 50
        },
        "pair_whitelist": [
            "ETH/BTC"
        ],
        "pair_blacklist": [
            "DOGE/BTC",
            "BNB/BTC"
        ]
    },
    "pairlists": [
        {
            "method": "StaticPairList"}
        ],
    "edge": {
        "enabled": false,
        "process_throttle_secs": 3600,
        "calculate_since_number_of_days": 7,
        "capital_available_percentage": 0.5,
        "allowed_risk": 0.01,
        "stoploss_range_min": -0.01,
        "stoploss_range_max": -0.1,
        "stoploss_range_step": -0.01,
        "minimum_winrate": 0.60,
        "minimum_expectancy": 0.20,
        "min_trade_number": 10,
        "max_trade_duration_minute": 1440,
        "remove_pumps": false
    },
    "telegram": {
        "enabled": true,
        "token": "kk",
        "chat_id": "jj"
    },
    "initial_state": "running",
    "forcebuy_enable": false,
    "internals": {
        "process_throttle_secs": 5
    }
}
```

 
### Observed Results:

freqtrade edge --strategy ProvaEdge --config configProvaEdge.json --stoplosses=-0.01,-0.2,-0.001 --timerange 20190131-20191210

| pair    |   stoploss |   win rate |   risk reward ratio |   required risk reward |   expectancy |   total number of trades |   average duration (min) |
|:--------|-----------:|-----------:|--------------------:|-----------------------:|-------------:|-------------------------:|-------------------------:|
| ETH/BTC |     -0.114 |       0.79 |                0.99 |                   0.26 |         0.57 |                      129 |                      651 |

freqtrade backtesting --strategy ProvaEdge --config configProvaEdge.json --timerange 20190131-20191210

| pair    |   buy count |   avg profit % |   cum profit % |   tot profit BTC |   tot profit % | avg duration   |   profit |   loss |
|:--------|------------:|---------------:|---------------:|-----------------:|---------------:|:---------------|---------:|-------:|
| ETH/BTC |         160 |          -0.24 |         -38.32 |      -0.00383570 |          -0.38 | 15:13:00       |       97 |     63 |
| TOTAL   |         160 |          -0.24 |         -38.32 |      -0.00383570 |          -0.38 | 15:13:00       |       97 |     63 |

| Sell Reason   |   Count |
|:--------------|--------:|
| sell_signal   |     160 |

As you can see freqtrade edge results are totally different from freqtrade bcktesting results:
1) in backtest we have 160 trades vs 129 trades in edge simulation
2) in backtest it is a losing strategy with a negative average profit, in edge it is a winning strategy with a positive expectancy 

i think that the results have to be similar in the two cases, because this simple strategy works only with buy and sell signals, not ROI, not trailingstop and the stoploss of the strategy is set to -0.114 (the same obtained with the edge simulation).
I tried with different strategies, different timeranges, different pairs, different configurations but the results between edge and backtesting are always different: always less trades and better expectancy in edge simulation.

Please, it is very important bug to solve, now it is better not to use these functions as they are unreliable as they can evaluate completely losing strategies as profitable!

Thanks for your help
Michele

**Comment by xmatthias** [comment]
First of all, thanks for reporting this in this detail, your instructions made it easy to confirm your findings.

Now i'm not the expert on Edge - i've rarely ever used it myself, so most i know is from reviewing and design discussions.
but letme try anyway:

Edge filters trades based on certain criteria and calculations - and each setting in the configuration is important - as that can remove a trade from the result.

simple example:
by increasing `max_trade_duration_minute` by 1000 (to 2440) - i get 155 trades, instead of the 129 you had.

a bit of tuning will get me to 160 trades eventually (not necessarily the same trades).

... To be continued

**Comment by Michele-GitHub** [comment]
You are right, the problem was that max_trade_duration_minute cut many losing strategies. I tested with a very high value of max_trade_duration_minute and now the results in backtesting and in edge are the same.

Problem selved, thankyou for your help!

**Comment by xmatthias** [comment]
hm - apparently i didn't finish my analysis - as my results did not line up yet ...

but glad i could help you 👍

**Comment by Michele-GitHub** [comment]
ok, let me know the details when you have finished the analysis...
thanks

**Comment by xmatthias** [comment]
well you said that the results in backtesting and edge are the same... so not sure what to analyze further ... ?

**Comment by Michele-GitHub** [comment]
yes my results now are right...i have understood you want to do more analysis.
ok task completed 
thanks
